### ISSUE TITLE
Work blocked on pollable object

### ISSUE THREAD

**Issue by pdunaj** [zephyrproject-rtos/zephyr#10749]
At the moment Zephyr has two kinds of work: executed straightaway and executed with a delay. It would be useful to add a third kind - work executed when provided pollable object is ready (or any object from the given set).

This would be useful in cases when user has a need to wait for a blocking API (e.g. read) but would like to avoid creating a dedicated thread to handle blocking (that would cost RAM used for the thread's stack).

This feature could be implemented by creating a thread that would wait for any of the pollable object to be ready and when it is it would submit an associated work to be executed.
It may be however better to create a hook in the scheduler and when a pollable object is ready submit an associated work there (e.g. much like unblocking the thread waiting for mutex).

**Comment by pdunaj** [comment]
Duplicate of https://github.com/zephyrproject-rtos/zephyr/issues/10748
